### PR TITLE
[issue 10693] use unify configuration from remote cluster for geo-replicator

### LINKED ISSUE #10693: Support config remote clusters' TLS settings for geo-replicator
This issue is similar to #10242

**Is your feature request related to a problem? Please describe.**
Currently the pulsarClient for geo-replicator using the local cluster's TLS configuration in broker.conf. If the remote cluster is using a different TLS certificate, the geo-replicator will fail to connect.

**Describe the solution you'd like**
Add TLS parameters for each remote cluster to `ClusterData`, when adding a remote cluster, the TLS settings of the cluster could be configured.

### PR DESCRIPTION
**Pull request by linlinnn**
Fixes #10693

### Motivation
use unify configuration from remote cluster for geo-replicator

### CODE REVIEW

**Comment by wangjialing218** [comment]
I have two suggestions:

1. There are too many paramaters for tls setting in `CmdClusters`. Could we combine all remote cluster auth setting to one paramater as a JSON string, just like "--external-pulsars" in `CmdFunctions`
https://github.com/apache/pulsar/blob/5a7ba52193c069bc40705bcdc74287aae1066b17/pulsar-client-tools/src/main/java/org/apache/pulsar/admin/cli/CmdFunctions.java#L325

2. Compared with `AuthenticationConfig` class, `tlsHostnameVerificationEnable` is missed. Shoud this paramter also be config in geo-replicator?
https://github.com/apache/pulsar/blob/6704f12104219611164aa2bb5bbdfc929613f1bf/pulsar-common/src/main/java/org/apache/pulsar/common/functions/AuthenticationConfig.java#L35

**Review by codelipenghui — APPROVED** [review]

**Comment by codelipenghui** [comment]
@linlinnn Could you please check the last comment of @wangjialing218 ?

File: pulsar-client-tools/src/main/java/org/apache/pulsar/admin/cli/CmdClusters.java [hunk 1]
```diff
@@ -83,11 +83,67 @@ void run() throws PulsarAdminException {
...
+
+        @Parameter(names = "--tls-trust-store", description = "TLS TrustStore path for internal client", required = false)
+        private String brokerClientTlsTrustStore;
+
+        @Parameter(names = "--tls-trust-store-pwd", description = "TLS TrustStore password for internal client", required = false)
+        private String brokerClientTlsTrustStorePassword;
+
+        @Parameter(names = "--tls-trust-certs-filepath", description = "path for the trusted TLS certificate file", required = false)
+        private String brokerClientTrustCertsFilePath;
+
```
Comment (gaoran10) [inline pulsar-client-tools/src/main/java/org/apache/pulsar/admin/cli/CmdClusters.java:106]: Maybe we could add a new base class for Create and Update command, there are many same configurations.

**Review by gaoran10 — COMMENTED** [review]
Great work!

**Comment by linlinnn** [comment]
@wangjialing218 @codelipenghui @gaoran10 I address your comments, could you please help review again?

**Comment by linlinnn** [comment]
@wangjialing218 
Thanks for your suggestions.
I use `--cluster-config-file` to solve your first suggestion.
For the second one, I don't see it in former logic.

**Review by gaoran10 — APPROVED** [review]
LGTM

**Pull request merged**